pandas/core/arrays/categorical.py

```diff
@@ -379,6 +382,109 @@ def __init__(
 
+    @classmethod
+    def from_dummies(
+        cls, dummies: "DataFrame", ordered: Optional[bool] = None
+    ) -> "Categorical":
+        """Create a `Categorical` using a ``DataFrame`` of dummy variables.
+
+        The ``DataFrame`` must be coercible to boolean,
+        and have no more than one truthy value per row.
+        The columns of the ``DataFrame`` become the categories of the `Categorical`.
+        A column whose header is NA will be dropped;
+        any row with a NA value will be uncategorised.
+
+        Parameters
+        ----------
+        dummies : DataFrame of bool-like
+        ordered : bool
+            Whether or not this Categorical is ordered.
+
+        Raises
+        ------
+        ValueError
+            If a sample belongs to >1 category
+
+        Returns
+        -------
+        Categorical
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(
+        ... [[1, 0, 0], [0, 1, 0], [0, 0, 1]], columns=["a", "b", "c"]
+        ... )
+        >>> Categorical.from_dummies(df)
+        [a, b, c]
+        Categories (3, object): [a, b, c]
+        """
+        df = dummies.drop(columns=np.nan, errors="ignore").astype(bool)
+
+        if (df.sum(axis=1) > 1).any():
+            raise ValueError("Some rows belong to >1 category")
+
+        mult_by = np.arange(df.shape[1]) + 1
+        #  000            000    0   -1
+        #  010            020    2    1
+        #  001 * 1,2,3 => 003 -> 3 -> 2 = correct codes
+        #  100            100    1    0
+        codes = (df * mult_by).sum(axis=1) - 1
+        codes[codes.isna()] = -1
+        return cls.from_codes(codes, df.columns.values, ordered=ordered)
+
+    def to_dummies(self, na_column=None) -> "DataFrame":
+        """
+        Create a ``DataFrame`` of boolean dummy variables representing this object.
+
+        For more power over column names or to use a sparse matrix,
+        see :func:`pandas.get_dummies`.
+
+        Parameters
+        ----------
+        na_column : Optional
+            If None, NA values will be represented as a row of zeros.
+            Otherwise, this is the name of a new column representing
+            those NA values.
+
+        Returns
+        -------
+        DataFrame
+
+        Examples
+        --------
+        >>> Categorical(["a", "b", "c"]).to_dummies()
+           a      b      c
+        0  True   False  False
+        1  False  True   False
+        2  False  False  True
+
+        >>> Categorical(["a", "b", np.nan]).to_dummies()
+           a      b
+        0  True   False
+        1  False  True
+        2  False  False
+
+        >>> Categorical(["a", "b", np.nan]).to_dummies("other")
+           a      b      other
+        0  True   False  False
+        1  False  True   False
+        2  False  False  True
+
+        See Also
+        --------
+        :func:`pandas.get_dummies`
+        """
+        from pandas import DataFrame, CategoricalIndex, Series
```

Comment:
Why is this not using the existing `get_dummies` implementation? This shold be identical to `pd.get_dummies(Categorical)`, right?